### ISSUE TITLE
PBFT signature count check is a bottleneck

### ISSUE THREAD

**Issue by nc6** [input-output-hk/ouroboros-network#877]
```
  let totalSigners = Seq.length signers
              gkSigners = Seq.length (Seq.filter (== gk) signers)
```

This is done on every header, and the length of signers can be pretty large (the size of the window + 2k). Testing shows this to be a major source of time in validating headers:

> 24.8     -     -                                                    Ouroboros.Consensus.Protocol.PBFT applyChainState.gkSigners (17265)

We should make this faster.

**Comment by nc6** [comment]
Also address https://github.com/input-output-hk/ouroboros-network/issues/876 as part of this.

**Comment by nc6** [comment]
Addressed as #893
